src/env.cc

```diff
@@ -348,8 +348,6 @@ Environment::Environment(IsolateData* isolate_data,
   credentials::SafeGetenv("NODE_DEBUG_NATIVE", &debug_cats, this);
   set_debug_categories(debug_cats, true);
 
-  isolate()->GetHeapProfiler()->AddBuildEmbedderGraphCallback(
-      BuildEmbedderGraph, this);
```

Comment:
I feel like this was kept in the right place in the Environment constructor … it should be added as early as possible, right? I mean, it probably doesn’t make a huge difference, but I think this should at least be added before the inspector is set up?

Reply:
Yeah, I think it makes sense to add it before the inspector is setup. I moved it out of the environment constructor because the constructor is supposed to be run in the snapshot builder, and it does not make sense to do this when building the snapshot. It would also be more coherent if all profiler/debugger stuff are done in one place.